backend/app/Repositories/UserRepository.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories;
+
+use App\Entities\User;
+use App\Repositories\Contracts\UserRepository as IUserRepository;
+
+final class UserRepository implements IUserRepository
```

Comment:
Name repository as EloquentUserRepository, such as we are going to have also ElasticSearchRepository. It is best practice not to call implementation different from interface